lava/magma/core/model/c/model.py

```diff
@@ -1,11 +1,49 @@
 # Copyright (C) 2021 Intel Corporation
 # SPDX-License-Identifier:  BSD-3-Clause
-from abc import ABC, abstractmethod
+from abc import ABC, abstractmethod, abstractproperty
 
-from lava.magma.core.model.model import AbstractProcessModel
+from numpy.distutils.core import setup
+from numpy.distutils.misc_util import Configuration, get_info
 
+from importlib import import_module, invalidate_caches
 
-class AbstractCProcessModel(AbstractProcessModel, ABC):
+import numpy as np
+from typing import List, Callable
+
+from lava.magma.core.model.py.model import AbstractPyProcessModel
+
+AbstractCProcessModel = None
+
+
+class CProcessModelMeta(type):
+    """
+    Compiles the sources specified in the class definition and then generates a type that includes the custom module in its base classes
+    """
+
+    def __new__(cls, name, bases, attrs):
+        if AbstractCProcessModel and AbstractCProcessModel in bases:
+
+            def configuration(parent_package="", top_path=None):
+                config = Configuration("", parent_package, top_path)
+                config.add_extension(
+                    "custom",
+                    attrs["source_files"],
+                    extra_info=get_info("npymath"),
+                )
+                return config
+
+            setup(
+                configuration=configuration,
+                script_args=["build_ext", "--inplace"],
+            )
+            invalidate_caches()
+            module = import_module("custom")
+            # from custom import Custom
+            bases += (module.Custom, AbstractPyProcessModel)
+        return super().__new__(cls, name, bases, attrs)
```

Comment:
```
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

This requires some judo I think, but it should be resolvable. Another possibility is to just compile the object in a separate function and then do manual type creation with `type(...)` rather than the metaclass.